addons/account/models/account_bank_statement.py

```diff
@@ -1226,6 +1236,13 @@ def reconcile(self, lines_vals_list, to_check=False):
 
             (line + counterpart_line).reconcile()
 
+        # Assign partner if needed (for example, when reconciling a statement
+        # line with no partner, with an invoice; assign the partner of this invoice)
+        if not self.partner_id:
+            rec_overview_partners = set(overview['partner_id'] for overview in reconciliation_overview)
+            if len(rec_overview_partners) == 1:
+                self.line_ids.write({'partner_id': rec_overview_partners.pop()})
```

Comment:
The keys of overview are currently `line_vals`/`counterpart_line`/`payment_vals`. So I suggest to remove the injection of `partner_id` and make this instead:
```
partners = set(overview['counterpart_line'].partner_id for overview in reconciliation_overview if overview['counterpart_line'] and overview['counterpart_line'].partner_id)
```
What do you think? :) (ofc feel free to indent this looooooooooong inlined line of code)

Reply:
Right, I missed that one. Thanks :)